woocommerce/woocommerce-ios
# [Mobile Payments] Update logic, strings on account related onboarding views

Closes: #5593 

@ctarda @joshheald - just one review needed but either of y'all are welcome

### Description
- Update comments in `WCPayAccountMapper` to reflect that data as an empty array hasn't been the case since WCPay plugin version 2.9.0 (Aug 2021)
- Pass plugin enum into setup-not-completed (i.e. no-account) view so we can prompt user to finish setting up the appropriate plugin
- Make review, rejected, country-not-supported views generic (i.e. don't mention WCPay) to match Android impl
- Don't display generic errors when we have an active plugin with no account - display setup-not-completed/no-account instead
- Update unit test to match that change

### Testing instructions
- Use the WCPay Dev Tools to "force the plugin to act as disconnected from WCPay" - this is the easiest (and non destructive) way to have your test site appear to have no account setup.
- Ensure you are prompted to finish account setup for the WCPay plugin
- Switch to a Stripe powered store
- Remove your keys from stripe plugin settings
- Ensure you are prompted to finish account setup for the Stripe plugin

### Screenshots

<img src="https://user-images.githubusercontent.com/1595739/150883898-9895a1c2-c57c-4a90-9ae8-2f5a24a71331.jpg" width=50% />

---
